### PR TITLE
 export hangup cause and put it in CDR reason/description 

### PR DESCRIPTION
**Pull request by eprueves**
This gets the [hangup_cause value](https://wiki.freeswitch.org/wiki/Hangup_Causes) from the FS xml cdrs and includes them in the cdr description. 
Our intention for this is to get more info from calls, especially those zero-second calls that we see in the CDRs. This might be useful for others as well. 

CDR reason/description will now be in this format:
```
2 sec call to 639361234567 (outside_call/NORMAL_CLEARING)
0 sec call to 639361234567 (local_call/USER_BUSY)
0 sec call to 639361234567 (outside_call/RECOVERY_ON_TIMER_EXPIRE)
```

### CODE REVIEW

**Comment by 9muir** [comment]
this looks useful, thanks.

File: client/core/federer_handlers/cdr.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def process_cdr(self, cdr_xml):
                 break
         if callee[0] == "+":
             callee = callee[1:]
+        hangupcause = get_tag_text(cdr_dom.getElementsByTagName("hangup_cause")[0].childNodes)
```
Comment (kkroo) [inline client/core/federer_handlers/cdr.py]: Thanks for your contribution-- and for patiently waiting for a review
Please define this in a helper function that returns a default string "UNKNOWN" in <1% error case the field is missing and you get a list index error. We should still process the billing event regularly

**Review by kkroo — CHANGES_REQUESTED** [review]
Thanks for your contribution-- and for patiently waiting for a review
Please define this in a helper function that returns a default string "UNKNOWN" in <1% error case the field is missing and you get a list index error. We should still process the billing event regularly

**Comment by eprueves** [comment]
I modified the code based on the PR review and pushed those edits to this branch. 
Thanks!

**Pull request closed without merging**